streaming-binlog/src/main/java/org/openmrs/analytics/FhirConverter.java

```diff
@@ -98,7 +105,17 @@ public void process(Exchange exchange) {
 			return;
 		}
 		String resourceJson = parser.encodeResourceToString(resource);
-		fhirStoreUtil.uploadResourceToCloud(resource.fhirType(), resource.getId(), resourceJson);
+		if (parquetUtil.getParquetPath() != null) {
+			try {
+				final ParquetWriter<GenericRecord> parquetWriter = parquetUtil.getWriter(resource.fhirType());
+				parquetWriter.write(parquetUtil.convertToAvro(resource));
```

Comment:
Do we want to expose the parquetWriter config options here? e.g pageSize, writerVersion, compression, e.t.c If defaults are ok, we can leave it at that

Reply:
Thanks for creating this ticket! Let's have the need for it before we implement it. From my experience with parquet,  having large block/page sizes impacts the performance since it requires more memory to manage. Couldn't find a good reference for this but this explains why [https://www.dremio.com/tuning-parquet/](https://www.dremio.com/tuning-parquet/). let's revisit this should we have any issue